packer/ansible/roles/node/tasks/main.yml

```diff
@@ -1,15 +1,21 @@
 ---
-- name: Download n Installer
-  get_url: url=http://git.io/n-install dest=/tmp/n-install mode=0755
-  register: download_n
-  sudo: yes
 
-- name: Install n
-  shell: "yes | N_PREFIX={{ ansible_env.HOME }}/n /tmp/n-install"
-  sudo: yes
-  when: download_n|changed
+# source ~/.bashrc is for interactive shell
+- name: Install nvm and Appending source string to ~/.bashrc
+  shell: wget -qO- https://raw.githubusercontent.com/creationix/nvm/v0.31.0/install.sh | bash
```

Comment:
@benbp actually, originally 'node' role is not used, and 'n' tool is also not used , it seems that it was used before then it's disabled or not used finally,do you know the history?

for nvm installation globally, nvm provide some parameters to install it to globally, and someone also have fork 'nvm' to install node and npm globally https://github.com/xtuple/nvm

'nvm' want to reduce the impacts to others and limit the node and npm version to local user, so if want to let others use node, we could install it globally.  .bashrc update is only for interactive shell to run command link 'vagrant ssh -c 'nf start'' 

for why not use' n install' , the problem is that 'n install' only install node, didn't install npm, npm still keep the same with before in system, maybe newer or older, not sure about that,  the newer version will have some problem for 'npm install' . so I choose 'nvm install' to keep consistent with travisCI for 'nvm' ,also for both node and npm, it will help use isolate the problems that caused by version problem, either Jenkins, or TravisCI, or users report. using 'n install is also fine, just need to additional  steps or docs to indicate npm version, and change travisCI to keep consistent with steps here. at least for version for both node and npm.

How do you think?